hs-ucan/library/Web/UCAN.hs

```diff
@@ -1,25 +1,24 @@
 module Web.UCAN
-  ( fromRawContent
+  ( parse
   , module Web.UCAN.Types
   ) where
 
 import           Data.Aeson
+
 import           RIO
+import qualified RIO.Text as Text
 
 import           Web.UCAN.Resolver.Error as Resolver
 import           Web.UCAN.Types
 
-
-fromRawContent ::
+parse ::
   ( FromJSON fct
   , FromJSON rsc
   , FromJSON ptc
   )
-  => RawContent
+  => ByteString
   -> Either Resolver.Error (UCAN fct rsc ptc)
-fromRawContent rawContent =
-  case decodeStrict rawContentBS of
-    Nothing   -> Left (InvalidJWT rawContentBS)
-    Just ucan -> Right ucan
-  where
-    rawContentBS = encodeUtf8 (unRawContent rawContent)
+parse bs =
+  case eitherDecodeStrict bs of
+    Left reason -> Left $ InvalidJWT (Text.pack reason) bs
+    Right ucan  -> Right ucan
```

Comment:
This is the big change. The awkward thing about this function is that it's now basically `Aeson.decode`. We do embed the value in `InvalidJWT`, so I think it probably does warrant its own function, but we can also drop this function and do that at the call sites of preferred.

Reply:
I need to write up some thoughts I have on using type classes for en/decode functions and why I dislike them :D

I love having specific functions like these :)

Reply:
Yeah, I like having both, because of canonical. That said, being able to abstract over things with To/FromJSON instances is hugely useful in practice.

I'd love to know why you avoid them!